lib/src/main/java/io/ably/lib/objects/batch/BatchContextBuilder.java

```diff
@@ -0,0 +1,16 @@
+package io.ably.lib.objects.batch;
+
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A functional interface for building and handling a BatchContext.
+ */
+@FunctionalInterface
+public interface BatchContextBuilder {
```

Comment:
Let's remove batch for now, it's not in the scope for this quarter, can be a bit tricky to implement in java, can distract a little

Reply:
removed